kafka-producer-application/kafka/README.md

```diff
@@ -0,0 +1,163 @@
+
+The KafkaProducer class implements multiple `send` methods, allowing the caller to provide `Callback` behavior once
+the event is sent to a Kafka topic. This tutorial will cover examples of both implementations.
+
+## Create a Producer and ProducerRecord
+
+There are required properties needed to create a Kafka Producer. At a minimum, the `Producer` needs to know:
+* How to find the Kafka broker(s).
+* How to serialize the key and value of events.
+
+```java annotate
+  final String bootstrapServers = "localhost:9092";
+  Properties properties = new Properties() {{
+      put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+      put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+      put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+      put(ProducerConfig.ACKS_CONFIG, "1");
+  }};
+  
+  Producer<String, String> producer = new KafkaProducer<>(properties);
+```
+
+For more about Kafka Producer Configurations, please [refer to the documentation](https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html).
+
+To prepare an event to send, we create instances of `ProducerRecord` - which include the target topic, key and value, of the event
+to be sent to Kafka.
+
+```java annotate
+    final String key = "...";
+    final String value = "...";
+    final String topic = "...";
+    ProducerRecord<String, String> record = ProducerRecord<>(topic, key, value);
+```
+
+## Send, No Callback
+
+Given our `ProducerRecord`, let's send this record via our `Producer`. 
+
+```java annotate
+    Future<RecordMetadata> result = producer.send(record);
+```
+
+## Send With Callback
+
+There are times when an application may want a deeper understanding of what occurred when an event was sent to Kafka. In this
+case we can utilize the implementation of the `send` method with a `Callback` function. 
+
+The `Callback` provides a way of handling any actions you want to take on request completion asynchronously. Note that the 
+`Callback` code executes on the producer’s I/O thread and any time consuming tasks could cause a delay in sending new 
+records, so any code here should be designed to execute quickly.
+
+Here is an example of a `Callback` - implemented as a lambda - that prints the offset and partition of the event sent to Kafka.
+
+```java annotate
+    Callback callback = (recordMetadata, e) -> 
+      System.out.println("topic: " + recordMetadata.topic() + 
+              "partition: , " + recordMetadata.partition() +
+              "offset: , " + recordMetadata.offset());
+
+    // use the callback with the `send` method...
+    Future<RecordMetadata> result = producer.send(record, callback);
+```
+
+## Using Example
+
+<details>
+<summary>Execute the Unit Tests</summary>
+
+There are junit test cases in this repo, exercising examples of both implementations of the `send` method of `KafkaProducer`.
```

Comment:
"There are test cases in this repo" makes sense if someone got to the tutorial by cloning the repo, but if viewing this on Confluent Developer it'd be confusing. CD users might not even know that a README in a GH repo is backing the page they're reading. Maybe step 1 for all options could tell them to clone if needed and `cd` to the right spot:

Clone the [confluentinc/tutorials](https://github.com/confluentinc/tutorials) GutHub repository if you haven't already:
```
git clone git@github.com:confluentinc/tutorials.git
```
And navigate to this tutorial's directory:
```
cd kafka-producer-application/kafka/
```

Reply:
Added this as preamble to the unit test section. Have a look and provide any feedback.